lib/Sema/CSApply.cpp

```diff
@@ -181,6 +181,21 @@ Solution::resolveConcreteDeclRef(ValueDecl *decl,
   auto sig = decl->getInnermostDeclContext()->getGenericSignatureOfContext();
   auto subst = computeSubstitutions(sig, locator);
 
+  // Lazily instantiate function definitions for class template specializations.
+  // Members of a class template specialization will be instantiated here (not
+  // when imported). If this method has already be instantiated, then this is a
+  // no-op.
+  if (auto *constMethod =
+          dyn_cast_or_null<clang::CXXMethodDecl>(decl->getClangDecl())) {
+    auto method = const_cast<clang::CXXMethodDecl *>(constMethod);
```

Comment:
Why is the `const_cast` required?  The destination isn't cont.

Reply:
The source is const. Updated to make this `auto const *`.